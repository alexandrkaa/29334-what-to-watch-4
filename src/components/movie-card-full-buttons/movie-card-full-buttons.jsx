import React from 'react';
import PropTypes from 'prop-types';
import {NavLink} from 'react-router-dom';

const MovieCardFullButtons = (props) => {
  const {onPlay, movieId, isReviewVisible} = props;
  return (
    <div className="movie-card__buttons">
      <button onClick={onPlay} className="btn btn--play movie-card__button" type="button">
        <svg viewBox="0 0 19 19" width="19" height="19">
          <use xlinkHref="#play-s"></use>
        </svg>
        <span>Play</span>
      </button>
      <button className="btn btn--list movie-card__button" type="button">
        <svg viewBox="0 0 19 20" width="19" height="20">
          <use xlinkHref="#add"></use>
        </svg>
        <span>My list</span>
      </button>
      {
        isReviewVisible &&
        <NavLink to={`/films/${movieId}/review`} className="btn movie-card__button">Add review</NavLink>
      }
    </div>
  );
};

MovieCardFullButtons.defaultProps = {
  isReviewVisible: false,
};

MovieCardFullButtons.propTypes = {
  onPlay: PropTypes.func.isRequired,
  movieId: PropTypes.number.isRequired,
  isReviewVisible: PropTypes.bool,
};

export default MovieCardFullButtons;
