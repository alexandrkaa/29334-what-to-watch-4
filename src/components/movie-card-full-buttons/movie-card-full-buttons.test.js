import React from 'react';
import renderer from 'react-test-renderer';
import {MovieCardFullButtons} from './movie-card-full-buttons.jsx';
import {BrowserRouter} from 'react-router-dom';

const onPlay = jest.fn();
const movieId = 10;
const isAuthorized = true;
const addToUserFavoriteList = jest.fn();
const removeFromUserFavoriteList = jest.fn();
const history = {};

describe(`<MovieCardFullButtons /> should render correctly`, () => {
  it(`<MovieCardFullButtons /> should render correctly then main page and favorite`, () => {
    const isMainPage = true;
    const isFavorite = true;
    const tree = renderer
      .create(
          <BrowserRouter>
            <MovieCardFullButtons
              onPlay={onPlay}
              movieId={movieId}
              isAuthorized={isAuthorized}
              isInUserFavoriteList={isFavorite}
              addToUserFavoriteList={addToUserFavoriteList}
              removeFromUserFavoriteList={removeFromUserFavoriteList}
              isMainPage={isMainPage}
              history={history}
            />
          </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`<MovieCardFullButtons /> should render correctly then not main page and not favorite`, () => {
    const isMainPage = false;
    const isFavorite = false;
    const tree = renderer
      .create(
          <BrowserRouter>
            <MovieCardFullButtons
              onPlay={onPlay}
              movieId={movieId}
              isAuthorized={isAuthorized}
              isInUserFavoriteList={isFavorite}
              addToUserFavoriteList={addToUserFavoriteList}
              removeFromUserFavoriteList={removeFromUserFavoriteList}
              isMainPage={isMainPage}
              history={history}
            />
          </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

