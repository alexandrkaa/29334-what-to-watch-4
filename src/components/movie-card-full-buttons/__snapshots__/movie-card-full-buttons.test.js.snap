// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCardFullButtons /> should render correctly <MovieCardFullButtons /> should render correctly then main page and favorite 1`] = `
<div
  className="movie-card__buttons"
>
  <button
    className="btn btn--play movie-card__button"
    onClick={[MockFunction]}
    type="button"
  >
    <svg
      height="19"
      viewBox="0 0 19 19"
      width="19"
    >
      <use
        xlinkHref="#play-s"
      />
    </svg>
    <span>
      Play
    </span>
  </button>
  <button
    className="btn btn--list movie-card__button"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="20"
      viewBox="0 0 19 20"
      width="19"
    >
      <use
        xlinkHref="#in-list"
      />
    </svg>
    <span>
      My list
    </span>
  </button>
</div>
`;

exports[`<MovieCardFullButtons /> should render correctly <MovieCardFullButtons /> should render correctly then not main page and not favorite 1`] = `
<div
  className="movie-card__buttons"
>
  <button
    className="btn btn--play movie-card__button"
    onClick={[MockFunction]}
    type="button"
  >
    <svg
      height="19"
      viewBox="0 0 19 19"
      width="19"
    >
      <use
        xlinkHref="#play-s"
      />
    </svg>
    <span>
      Play
    </span>
  </button>
  <button
    className="btn btn--list movie-card__button"
    onClick={[Function]}
    type="button"
  >
    <svg
      height="20"
      viewBox="0 0 19 20"
      width="19"
    >
      <use
        xlinkHref="#add"
      />
    </svg>
    <span>
      My list
    </span>
  </button>
  <a
    aria-current={null}
    className="btn movie-card__button"
    href="/films/10/review"
    onClick={[Function]}
  >
    Add review
  </a>
</div>
`;
